src/Microsoft.IdentityModel.Tokens/LogMessages.cs

```diff
@@ -134,6 +134,7 @@ internal static class LogMessages
         // public const string IDX10622 = "IDX10622:";
         // public const string IDX10623 = "IDX10623:";
         // public const string IDX10624 = "IDX10624:";
+        public const string IDX10625 = "IDX10625: Failed to verify the length of the authenticationTag '{0}'.";
```

Comment:
Consider adding an aka.ms link that would point to an article on the [wiki](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki) explaining why one would start seeing this exception, how to set the config switch and what are the risks.
Log message seems misleading too, IdentityModel can verify, but verification fails as actual tag length doesn't correspond to the expected one.

Reply:
Adding the algorithm and actual/expected tag length to the message can also be helpful.

Reply:
Added aka.ms link with a wiki page, updated log message, added algorithm, an expected/actual tag lengths.